src/hmem.c

```diff
@@ -528,12 +528,74 @@ bool ofi_hmem_is_initialized(enum fi_hmem_iface iface)
 	return hmem_ops[iface].initialized;
 }
 
+#include <stdio.h>
+
+void ofi_hmem_set_iface_filter(const char* iface_filter_str, bool* filter)
+{
+	int iface, rlen, llen;
+	char* entry = NULL;
+	const char* token = ";";
+	const char* iface_labels[ARRAY_SIZE(hmem_ops)] = {
+		"system",   // FI_HMEM_SYSTEM
+		"cuda",     // FI_HMEM_CUDA
+		"rocr",     // FI_HMEM_ROCR
+		"ze",       // FI_HMEM_ZE
+		"neuron",   // FI_HMEM_NEURON
+		"synapseai" // FI_HMEM_SYNAPSEAI
+	};
+
+	/* initialize array */
+	memset(filter, false, sizeof(bool) * ARRAY_SIZE(hmem_ops));
+
+	/* always enable system hmem interface */
+	filter[FI_HMEM_SYSTEM] = true;
+
+	/* tokenize the string */
+	entry = strtok(iface_filter_str, token);
+	while (entry != NULL) {
+		llen = strlen(entry);
+
+		/* search the list of known hmem interfaces, switch to true if found */
+		for (iface = 0; iface < ARRAY_SIZE(hmem_ops); iface++) {
+			rlen = strlen(iface_labels[iface]);
+
+			/* if the length is the same, and entry matches the iface
+			 * string (case-insensitive), then enable the iface in the filter
+			 */
+			if (llen == rlen && !strncasecmp(iface_labels[iface], entry, llen)) {
```

Comment:
Could just use `strcasecmp()` to simplify the logic.

Reply:
I'll make the switch. I tend to aim toward defensive strategies when dealing with user input.

Reply:
`strncasecmp` is not more defensive than `strcasecmp` here since (1) the prior call to `strlen` would have encountered the same risk, and (2) the string `entry` is known to have limited length.